cloudevents/working-drafts/dds-protocol-binding.md

```diff
@@ -0,0 +1,265 @@
+interested in. This reduces the amount of data that is sent over the wire in a
+DDS-based system.
+- Reliability and Durability: These Quality of Service (QoS) settings control
+how DDS handles reliable message delivery of data samples with the same key, and
+when to overwrite locally available data samples with the same key.
+
+While it is not necessary to set the `datakey` field for a DDS CloudEvent message,
+setting this value will enable many of the more powerful features of the DDS
+protocol.
+
+### 3.2. Binary Content Mode
+
+The _binary_ content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.2.1. Content Type
+
+For the _binary_ mode, the header `content-type` property MUST be mapped
+directly to the CloudEvents `datacontenttype` attribute and the
+`datacontentencoding` attribute MUST be set to `binary`.
+
+#### 3.2.2. Event Data Encoding
+
+The [`data`](#21-data) byte-sequence MUST be used as the body ("Data" field)
+of the DDS message.
+
+In binary mode, the DDS representation of a CloudEvent with no `data` is a
+DDS message with no body. Transmission such a DDS message is allowable.
+
+#### 3.2.3. Metadata Headers
+
+All [CloudEvents][ce] attributes and [CloudEvent Attributes Extensions]
+(../primer.md#cloudevent-extension-attributes)
+with exception of `data` MUST be individually mapped to and from the Header
+fields in the DDS message.
+
+### 3.3. Structured Content Mode
+
+The _structured_ content mode keeps event metadata and data together in the
+payload, allowing simple forwarding of the same event across multiple routing
+hops, and across multiple protocols.
+
+#### 3.3.1. Content-Type
+
+If present, the DDS message header property `content-type` MUST be set to the
+media type of an [event format](#14-event-formats).
+
+Example for the [JSON format][json-format]:
+
+```text
+content-type: cloudevent/json
+```
+Example for the [DDS event format][dds-message-format]:
+
+```text
+content-type: application/cloudevent+dds
+```
+
+For both of the above cases, the `content-type` property MUST be mapped
+directly to the CloudEvents `datacontenttype` attribute and the
+`datacontentencoding` attribute MUST be set to either `text` or `json`
+depending on the type of structured data being transmitted.
+
+
+#### 3.3.2. Event Data Encoding
+
+The chosen [event format](#14-event-formats) defines how all attributes, and
+`data`, are represented.
+
+The event metadata and data are then rendered in accordance with the
+[event format](#14-event-formats) specification and the resulting data becomes
+the DDS application [data](#21-data) section.
+
+Similar to binary mode, the DDS representation of a structured CloudEvent with
+no `data` is a DDS message with no body, and transmission of such a message
+is allowable.
+
+#### 3.3.3. Metadata Headers
+
+Implementations include the same DDS headers as defined for the
```

Comment:
I'm reading this as the CE attributes are to appear as DDS Headers AND in the CE structure too. Is that the intent?

Reply:
Currently we're really only supporting (reading) the content-type header for both binary and structured data, so I updated the text to just say this.